shared/types/field/myinfo/base.ts

```diff
@@ -0,0 +1,137 @@
+}
+
+export enum MyInfoDataSource {
+  HDB = 'Housing Development Board',
+  ICA = 'Immigration & Checkpoints Authority',
+  MOM = 'Ministry of Manpower',
+  URA = 'Urban Redevelopment Authority',
+  User = 'User-provided',
+  MSFD = 'Ministry of Social and Family Development',
+}
+
+const MYINFO_VERIFICATION_TYPE_MAPPINGS: {
+  [K in MyInfoVerifiedType]: VerifiedFor
+} = {
+  F: VerifiedFor.ForeignersWithSingpass,
+  PR: VerifiedFor.PermanentResidents,
+  SG: VerifiedFor.Singaporeans,
+}
+
+const extractVerifiedFor = (
+  myInfoFieldType: MyInfoAttribute,
+): VerifiedForMappings => {
+  const verificationArray = MYINFO_FIELD_CONSTANTS[myInfoFieldType]['verified']
+  const baseVerifiedFor = {
+    [VerifiedFor.Singaporeans]: false,
+    [VerifiedFor.PermanentResidents]: false,
+    [VerifiedFor.ForeignersWithSingpass]: false,
+  }
+
+  verificationArray.forEach((element) => {
+    baseVerifiedFor[MYINFO_VERIFICATION_TYPE_MAPPINGS[element]] = true
+  })
+
+  return baseVerifiedFor
+}
+
+const convertMyInfoDataSource = (source: string): MyInfoDataSource[] => {
+  return source
+    .split('/')
+    .map((dataSource) => dataSource.trim() as MyInfoDataSource)
+}
+
+// Type for MyInfo field within the editor view
+// A MyInfo field has UI related data tagged to it (description etc)
+// And it wraps a specific field type on public forms that is prefilled
+export type MyInfoFieldWithMeta<T extends MyInfoField = MyInfoField> = T &
+  MyInfoPreviewMeta
+
+// Type for MyInfo meta information shown on the editor sidebar
+// The data here is used for visual feedback to the user
+export type MyInfoPreviewMeta = {
+  dataSource: MyInfoDataSource[]
+  verifiedFor: {
+    [K in VerifiedFor]: boolean
+  }
+  details: string
+}
+
+// Type that has field level meta information (eg: validation)
+// This is used to set sensible defaults for MyInfo fields
+export type MyInfoFieldMeta<T extends MyInfoableFieldBase> = Omit<
+  T,
+  keyof FieldBase
+>
+
+export const isMyInfo = (field: FormField): field is MyInfoField => {
+  switch (field.fieldType) {
+    case BasicField.Date:
+    case BasicField.Dropdown:
+    case BasicField.HomeNo:
+    case BasicField.Mobile:
+    case BasicField.Number:
+    case BasicField.ShortText:
+      return !!field.myInfo
+    default:
+      return false
+  }
+}
+
+export const extendWithMyInfo = (field: MyInfoField): MyInfoPreviewMeta => {
+  return {
+    dataSource: convertMyInfoDataSource(
+      MYINFO_FIELD_CONSTANTS[field.myInfo.attr].source,
+    ),
+    verifiedFor: extractVerifiedFor(field.myInfo.attr),
+    details: MYINFO_FIELD_CONSTANTS[field.myInfo.attr].description,
+  }
+}
+
+export const myInfoTextFieldMeta: MyInfoFieldMeta<ShortTextFieldBase> = {
+  ValidationOptions: {
+    selectedValidation: null,
+    customVal: null,
+  },
+}
+
+export const getMyInfoDropdownFieldMeta = (
+  myInfoAttribute: MyInfoDropdownTypes,
+): MyInfoFieldMeta<DropdownFieldBase> => {
+  const myInfoDropdownFields = _(MYINFO_FIELD_CONSTANTS)
```

Comment:
i felt that this way (chaining) was cleaner rather than isolating the imports and writing it as 

```ts
const filteredFieldOptions = ... 
const myInfoDropdownFields = ...
return ...
```

but am open to changing if bundle size is an issue